File: 1929-concatenation-of-array/1929-concatenation-of-array.py
##time:o(n)
##space:o(n)
class Solution:
    def getConcatenation(self, nums: List[int]) -> List[int]:
        
        ans = []
        
        for i in range(2):
            for j in nums:
                ans.append(j)
        return ans
    
    
# Create an instance of the Solution class
#solution = Solution()

# Provide a list of integers as input
#nums = [1, 2, 3, 4, 5]

# Call the getConcatenation method and store the result
#result = solution.getConcatenation(nums)

# Print the concatenated list
#print("Concatenated list:", result)
    

File: 1929-concatenation-of-array/1929-concatenation-of-array.cpp
//time:o(n)
//space:o(n)
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        
        vector<int> ans;
        
        int len = nums.size();
        for(int i=0;i<2*len;i++){
            ans.push_back(nums[i%len]);
        }
        return ans;
        
    }
};

/*
int main() {
    Solution solution;
    vector<int> nums = {1, 2, 3, 4, 5};
    vector<int> result = solution.getConcatenation(nums);
    
    // Print the concatenated list
    for (int num : result) {
        cout << num << " ";
    }
    
    return 0;
}
*/